src/qibo/quantum_info/basis.py

```diff
@@ -0,0 +1,211 @@
+        matrix into 4 :math:`\frac{N}{2} \times \frac{N}{2}`
+        matrices, where :math:`N = 2^{n}` and :math:`n` is the
+        number of qubits.
+
+        Args:
+            matrix: :math:`N \times N` matrix.
+            nrows (int): number of rows of the block matrix.
+            ncols (int): number of columns of the block matrix
+
+        Returns:
+            Block-vectorization of ``matrix``.
+        """
+        dim, _ = matrix.shape
+        return (
+            matrix.reshape(int(dim / nrows), nrows, -1, ncols)
+            .swapaxes(1, 2)
+            .reshape(-1, nrows, ncols)[[0, 2, 1, 3]]
+        )
+
+    if backend is None:
+        from qibo.backends import GlobalBackend
+
+        backend = GlobalBackend()
+
+    d = len(state)
+    n = int(d / 2)
+    nqubits = int(np.log2(d))
+
+    if len(state.shape) == 1:
+        state = np.outer(state, state.conj())
+
+    if n == 1:
+        state = state.reshape((1, -1), order="F")[0]
+    else:
+        state = block_split(state, n, n)
+        for _ in range(nqubits - 2, 0, -1):
+            n = int(n / 2)
+            state = np.array([block_split(matrix, n, n) for matrix in state])
+            state = state.reshape((np.prod(state.shape[:-2]), *(state.shape[-2:])))
+        state = np.array(
+            [matrix.reshape((1, -1), order="F") for matrix in state]
+        ).flatten()
+
+    return backend.cast(state, dtype=state.dtype)
+
+
+def pauli_basis(
+    nqubits: int, normalize: bool = False, vectorize: bool = False, backend=None
+):
+    """Creates the ``nqubits``-qubit Pauli basis.
+
+    Args:
+        qubits (int): number of qubits.
+        normalize (bool, optional): If ``True``, normalized basis ir returned.
+            Defaults to False.
+        vectorize (bool, optional): If ``False``, returns a nested array with
+            all Pauli matrices. If ``True``, retuns an array where every
+            row is a vectorized Pauli matrix. Defaults to ``False``.
+        backend (``qibo.backends.abstract.Backend``, optional): Backend for execution.
+            If ``None``, defaults to ``GlobalBackend()``.
+
+    Returns:
+        list: list with all Pauli matrices forming the basis.
+    """
+
+    if nqubits <= 0:
+        raise_error(ValueError, "nqubits must be a positive int.")
+
+    if not isinstance(normalize, bool):
+        raise_error(
+            TypeError,
+            f"normalize must be type bool, but it is type {type(normalize)} instead.",
+        )
+
+    if not isinstance(vectorize, bool):
+        raise_error(
+            TypeError,
+            f"vectorize must be type bool, but it is type {type(vectorize)} instead.",
+        )
+
+    if backend is None:
+        from qibo.backends import GlobalBackend
+
+        backend = GlobalBackend()
+
+    basis = [
+        I(0).asmatrix(backend),
+        X(0).asmatrix(backend),
+        Y(0).asmatrix(backend),
+        Z(0).asmatrix(backend),
+    ]
```

Comment:
Here you can also use
```suggestion
    from qibo.matrices import I, X, Y, Z
    basis = [I, X, Y, Z]
```
Note that these matrices are always numpy arrays.

Reply:
I couldn't import `qibo.matrices`. Do you mean `from qibo.backends.npmatrices import NumpyMatrices` ?

Reply:
changed that import and removed backend from all `basis.py` functions